### ISSUE TITLE
Workspace scan freezes

### ISSUE THREAD

**Issue by borisdrovnin** [castwide/solargraph#293]
ruby version: 2.5.3
solargraph version: 0.38.4+
os: linux

The following example code freezes a workspace scan (`solargraph scan -v`) and causes 100% processor utilization.

```Ruby
@object = @object.a
@object = @object.b
@object = @object.c
@object = @object.d
@object = @object.e
@object = @object.f
@object = @object.g
@object = @object.h
@object = @object.i
@object = @object.g
```
Rollback to version 0.38.3 fixes the problem.

**Comment by castwide** [comment]
Confirmed. It looks like type inference goes into an infinite loop. I'll investigate further.

**Comment by castwide** [comment]
Fixed in the master branch. Slated for release in v0.38.6.

**Comment by castwide** [comment]
Released in v0.38.6.
